mpsort-mpi.c

```diff
@@ -517,13 +517,10 @@ MPIU_Gather (MPI_Comm comm, int root, const void * sendbuffer, void * recvbuffer
         rdispls[i] = rdispls[i - 1] + recvcount[i - 1];
     }
 
-    if(ThisTask == root) {
```

Comment:
Actually, the recv_buffer NULL case is the canonical use case of the function. providing a recvbuffer overrides the allocation behaviour. 

The reason is we compute the size of recv buffer inside MPIU_Gather -- if we require the user to allocate recvbuffer before calling MPIU_Gather, then the caller must compute the size, duplicating this logic of MPIU_Gather.

In this argument, I am just looking at this function, not how it is currently used in mp-sort. (remember MPIU_Gather is a copy & pastable function that intend to be used elsewhere).

Reply:
Oh, this I didn't know! I will revert that change.